stdio/bignum.c

```diff
@@ -104,21 +158,26 @@ static int bignum_cmpInternal(const bignum_t *first, const bignum_t *second)
 int bignum_init(bignum_t *bn, size_t size, bignum_data_t val)
 {
 	bignum_t result = { .data = NULL, .len = 1, .size = size };
-	result.data = calloc(result.size, sizeof(*result.data));
-	if (result.data != NULL) {
-		result.data[0] = val;
-		*bn = result;
-		return 0;
+	if (size > BIGNUM_STACK_BUFFER_SIZE) {
```

Comment:
To simplify the logic, how about:
- init bn->data = bn->stackBuffer,
- init bn->size = BIGNUM_STACK_BUFFER_SIZE
- do bignum_resize(bn, size).

sidenote - `result` temp var seems redundant to me